app/controllers/articles.server.controller.js

```diff
@@ -12,8 +12,17 @@ var mongoose = require('mongoose'),
  * Create a article
  */
 exports.create = function(req, res) {
+
+	// Filters out control fields
+	delete req.body.updatedAt;
+	delete req.body.updatedBy;
+	delete req.body.createdAt;
+	delete req.body.createdBy;
+
 	var article = new Article(req.body);
-	article.user = req.user;
+
+	article.createdBy = req.user;
```

Comment:
This will not work:

delete req.body.updatedAt;
delete req.body.updatedBy;
delete req.body.createdAt;
article.createdBy = req.user;
article.user = req.user;
var article = new Article(req.body);

as the new Article will create a new object that will substitute the old prepared article var. So I think you meant:
delete req.body.updatedAt;
delete req.body.updatedBy;
delete req.body.createdAt;
req.body.createdBy = req.user;
var article = new Article(req.body);

that will work indeed. My only hesitation about this change is that it kinda breaks the logic I tried to convey:

someFunction(inputObject) {
  // 1st part: some logic to clean the input, very generic, not specific to someFunction real work

  // 2nd part: the real work that someFunction must do:
}

In this case, the deletion on req.body properties is the 1st part, some generic logic to clean/check security:

        // Filters out control fields
	delete req.body.updatedAt;
	delete req.body.updatedBy;
	delete req.body.createdAt;
	delete req.body.createdBy;

then comes the second part, the one is specific to this function, the creation of a new Article:

	var article = new Article(req.body);

with the required logic to build a new Article, in this case, to ensure that we record the creator of the article:

	article.createdBy = req.user;
	article.updatedBy = req.user;

Also, I think adding data to the req.body object is going to have side effects if other middleware were to use the same req.body object later in the call chain.
I realize that the same could be said about the deletion of the fields I propose, I suppose, and I would agree on that and the only thing I could say is that those fields are internal to the app and could never be a valid parameter on the request payload or that we should be explicit in the properties of req.body that we'll use for the article creation ;). 
Something along the lines 
var article = new Article({
   firstName: req.body.firstName,
   lastName: req.body.lastName
});

would be the ultimate (but tedious) secure way to build an object using data from the user without fears of security implications or unintended side effects.

I hope this clarifies my implementation.